rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -157,9 +343,39 @@ def _decode_curve_remove_events(
         # 3. Withdrawal 2
         # etc.
         if return_event is None or len(withdrawal_events) == 0:
-            # for deposit zap contracts, this is handled during post decoding
-            if user_or_pool_address in CURVE_DEPOSIT_CONTRACTS:
-                return DecodingOutput(matched_counterparty=CPT_CURVE)
+            # we have the `return_event` but we need the withdrawal events.
+            # so we loop through all the logs searching for the ones that match.
+            if user_or_pool_address in CURVE_DEPOSIT_CONTRACTS and return_event is not None:
+                assets_to_amount: dict[Asset, int] = defaultdict(int)
+                for log_entry in all_logs:
+                    if (
+                        log_entry.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                        hex_or_bytes_to_address(log_entry.topics[1]) == user_or_pool_address and
+                        log_entry.address in CURVE_DEPOSIT_CONTRACTS[user_or_pool_address][0]
+                    ):
+                        assets_to_amount[Asset(ethaddress_to_identifier(log_entry.address))] = hex_or_bytes_to_int(log_entry.data[:32])  # noqa: E501
```

Comment:
Afaik we try to avoid using `ethaddress_to_identifier` in favor of `evm_address_identifier`